docs/configuration.md

```diff
@@ -1411,6 +1411,15 @@ Apart from these, the following properties are also available, and may be useful
   </td>
 </tr>
 <tr>
+  <td><code>spark.blacklist.application.fetchFailure.enabled</code></td>
+  <td>false</td>
+  <td>
+    (Experimental) If set to "true", Spark will blacklist the executors immediately when the fetch failure
+    happened. If external shuffle service is enabled, then the whole node will be blacklisted. This configuration
+    is to handle some scenarios where shuffle fetch is available and cannot be recovered through retry.
```

Comment:
the last sentence is pretty confusing ... I am not sure what else to put, maybe just leave it out?  

This change would be helpful when fetch failures come from non-transient issues, but would cause extra recomputations when the failures are transient.  Unfortunately that doesn't really give much guidance to the end user ... I feel like even ourselves we don't really have clear guidelines on when it should be set, its sort of an emergency escape-hatch we're leaving in place.